plugin/db/mysql/mysql.go

```diff
@@ -913,21 +915,31 @@ func (driver *Driver) Dump(ctx context.Context, database string, out io.Writer,
 }
 
 // Restore restores a database.
-func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error) {
+func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner, config db.RestoreConfig) (err error) {
 	txn, err := driver.db.BeginTx(ctx, nil)
 	if err != nil {
 		return err
 	}
 	defer txn.Rollback()
 
-	f := func(stmt string) error {
+	fnExecuteStmt := func(stmt string) error {
+		if config.IsGhostTable {
+			driver.l.Debug("original", zap.String("stmt", stmt))
+			// restore to ghost tables, need to rewrite SQL
+			stmt = rewriteToGhostTableForRestore(stmt, time.Now().Format("20060102150405"))
```

Comment:
It's great to have a timestamp suffix, however, I feel like this suffix should be part of config and passed in from callers because the same suffix will be used for applying binary logs.